### PR TITLE
Query and update animals table

### PR DESCRIPTION
**Pull request by cvalencia1991**
# In this milestone, I did the following activities:
- Update TABLE
- Using Transitions
- Using Groups in PostgreSQL
- Using Functions in PostgerSQL

### CODE REVIEW

**Comment by cvalencia1991** [comment]
Update the information that i form the table 

image

Inside a transaction:
- Update the animals table by setting the species column to digimon for all animals that have a name ending in mon.
- Update the animals table by setting the species column to pokemon for all animals that don't have species already set.
- Commit the transaction.
- Verify that change was made and persists after commit.

image

- Now, take a deep breath and... Inside a transaction delete all records in the animals table, then roll back the transaction.
- After the rollback verify if all records in the animals table still exists. After that, you can start breathing as usual ;)

image
image
image
image
image

## Inside a transaction:

image

- Delete all animals born after Jan 1st, 2022.

image

- Create a savepoint for the transaction.

image

- Update all animals' weight to be their weight multiplied by -1.

image

- Rollback to the savepoint

image

- Update all animals' weights that are negative to be their weight multiplied by -1.

image


- Commit transaction

image

image

How many animals are there?
image

How many animals have never tried to escape?

image

What is the average weight of animals?

image

Who escapes the most, neutered or not neutered animals?
- the neutered escapes 4 times and the not neutered escapes 2 times
image

What is the minimum and maximum weight of each type of animal?
image

What is the average number of escape attempts per animal type of those born between 1990 and 2000?

image

File: queries.sql [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+-- How many animals have never tried to escape?
+SELECT MIN(escape_attempts) FROM animals;
+-- What is the average weight of animals?
+SELECT AVG(weight_kg) FROM animals;
+-- Who escapes the most, neutered or not neutered animals?
+SELECT COUNT(neutered) FROM animals GROUP BY neutered;
+-- What is the minimum and maximum weight of each type of animal?
+SELECT  species,max(weight_kg),min(weight_kg) FROM animals GROUP BY species;
+-- What is the average number of escape attempts per animal type of those born between 1990 and 2000?
+SELECT name, AVG(escape_attempts) FROM animals WHERE  date_of_birth> '1990-01-01 00:00:00' AND date_of_birth < '2000-01-01 00:00:00' GROUP BY name;
```
Comment (t-yanick) [inline queries.sql]: - [OPTIONAL] The range for the date of birth should include the end on 2000, please kindly fix that.

**Review by t-yanick — APPROVED** [review]
Hi @cvalencia1991, 

### Great job !!!
### To Highlight! :clap:  :green_circle: 

:heavy_check_mark:  Good commit messages
:heavy_check_mark: Descriptive Readme file

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit:  
Congratulations! 🎉 

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

### N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: queries.sql (hunk 1 above)
Comment (cvalencia1991) [reply queries.sql]: Ok im going to do that

**Comment by cvalencia1991** [comment]
Thank you :smile:

**Pull request merged**